Implements notebook delete be/fe feature

kyucumber opened the pull request:
노트북 삭제 기능 be/fe 기능 구현

![](https://s3.ap-northeast-2.amazonaws.com/woowanotes3/static/%E1%84%89%E1%85%B3%E1%84%8F%E1%85%B3%E1%84%85%E1%85%B5%E1%86%AB%E1%84%89%E1%85%A3%E1%86%BA+2018-08-25+%E1%84%8B%E1%85%A9%E1%84%92%E1%85%AE+2.49.21.png)

Commit:
Implements blank notebook create bug fix feature #134

Dto를 이용해 노트북 생성하도록 수정, 에러메세지 및 @Valid를 이용한 validation 체크 추가
서비스 반영, 테스트 코드 수정사항 반영 및 validation 메세지 체크하는 테스트 코드 추가.

Commit:
Implements blank notebook create bug fix feature #134

프론트에서 노트북 추가 실패 시 에러메세지 노출하도록 수정
일부 css 수정, 이미지 형태의 버튼을 아이콘으로 수정(이미지 요청 api 요청이 가게 되므로 아이콘 형태로 변경)

Commit:
Implements notebook delete backend feature #138

노트북 삭제 BE 부분 구현

NoteBook entity에 delete 컬럼 추가, delete 메소드 추가
User entity lazy initialize exception 해결 위해 ToString에 lazy 컬럼 제거 후 재정의
User entity에 비영속 엔티티와의 비교를 위한 메소드 추가
NoteBookService에 노트북 삭제 코드 추가(유저 정보가 같아야만 삭제.)
노트북 삭제 api 생성 및 테스트 코드 작성
유저 정보가 다른 경우 UnAuthorizedException, ErrorDetails 형태 응답 반환해주도록 ControllerAdvice에 추가

Commit:
Implements notebook delete backend feature #138

delete = false인 노트북만 가져오도록 수정

Commit:
Merge branch 'feature/empty-notebook-validation' into feature/notebook-delete

* feature/empty-notebook-validation:
  Implements blank notebook create bug fix feature #134
  Implements blank notebook create bug fix feature #134

# Conflicts:
#	src/main/java/com/wannaone/woowanote/support/ErrorMessage.java
#	src/test/java/com/wannaone/woowanote/web/ApiNoteBookAcceptanceTest.java

Commit:
Refactor edit test fail

기존 실패 테스트 수정, notebookDto 사용하도록 수정, ownerIdAndDeletedFalse 사용하도록 변경

Commit:
Implements notebook delete feature fe #138

노트북 삭제 프론트 코드 작성 삭제 버튼 추가 css 적용
클릭 시 삭제 이벤트 관련 코드 작성

Commit:
Refactor

remove unnecessary import and space

Commit:
Refactor

edit notebook add button css

dain7672 commented on src/main/java/com/wannaone/woowanote/domain/User.java:
```diff
@@ -73,4 +71,26 @@ public String getEmail() {
     public String getPassword() {
         return password;
     }
+
```
신기한 함수가 있네요.

dain7672 commented on src/main/java/com/wannaone/woowanote/dto/NoteBookDto.java:
```diff
@@ -0,0 +1,19 @@
+    private String title;
+    public NoteBook toEntity(){
+        return new NoteBook(this.title);
+    }
```
문제 없기는 한데, #137 PR과 커밋이 겹치네요

dain7672 commented on src/main/java/com/wannaone/woowanote/service/NoteBookService.java:
```diff
@@ -20,16 +22,27 @@
+        }
+        deleteNoteBook.delete();
+        return deleteNoteBook;
+    }
```
👍 왠지 equals 문제로 고생했을듯..ㅋㅋ

kyucumber replied:
고민하다가.. 굳이 또 DB에서 꺼내서 영속상태로 만든다음에 비교할 필요는 없을거 같아서
그냥 ㅋㅋ.. LAZY인 부분 빼고 비교하게 해놨어

dooho-h replied:
owner id만 비교하면 안되나요?!

kyucumber replied:
ownerId만 비교할까?

dain7672 commented on src/main/resources/static/js/main-app.js:
```diff
@@ -84,7 +84,19 @@ class MainApp {
+        const target = e.target;
+        //노트북 삭제 버튼이 클릭된 경우
+        if(target.tagName === 'I' && confirm('해당 노트북을 삭제하시겠습니까?')) {
+            const successCallback = () => {
```
꼭 이렇게 하지 않아도, 삭제 버튼에 event를 주면 알아서 selectnotebookevent는 실행이 안되지 않을까요?

kyucumber replied:
삭제 버튼이 다 동적으로 추가돼서 일반적으로 걸면 안될거 같아서 이벤트 델리게이션으로 처리하다보니까 이렇게 되버렸는데
삭제버튼에 class나 이런 값을 주고 따로 이벤트를 거는게 나으려나? 그렇게 처리할 수 있으면 그게 좋을듯

dooho-h commented on src/main/java/com/wannaone/woowanote/domain/User.java:
```diff
@@ -73,4 +71,26 @@ public String getEmail() {
+                Objects.equals(password, user.password) &&
+                Objects.equals(name, user.name) &&
+                Objects.equals(photoUrl, user.photoUrl);
+    }
```
이거슨 무엇이오..?

kyucumber replied:
사실 로그인한 유저 세션에 들어간 유저 아이디 값만 맞으면 상관없을거 같긴 한데 아이디만 비교하게 바꿀까?

doooyeon replied:
넹 바꿔주세요 :)

kyucumber replied:
바꿨어~

dooho-h commented on src/main/java/com/wannaone/woowanote/domain/User.java:
```diff
@@ -73,4 +71,26 @@ public String getEmail() {
+                ", name='" + name + '\'' +
+                ", photoUrl='" + photoUrl + '\'' +
+                '}';
+    }
```
이것도 필요한가요?

kyucumber replied:
@ToString 달면 되는데 저러니까 Lazy인 애도 Tostring해주면서 자꾸 LAZY exception 나서 LAZY인 애는 빼고 ToString하게 직접 그냥 구현했어

doooyeon replied:
근데 우리 다 eager 아닌가요?
LAZY exception은 왜 나는지 ㅠㅠ

kyucumber replied:
@OneToMany는 default가 LAZY래..

Commit:
Merge branch 'develop' into feature/notebook-delete

* develop:
  공백 노트북 생성되지 않도록 수정 (#135)
  Implements image upload file size limit feature #144 (#145)
  Implements notebook first init issue #140 (#141)
  Implements note delete alert feature #136 (#137)
  Feature/travis deploy prod setting (#151)
  Implements Travis CI/CD #14 (#150)
  Implements add writer info when creating comments feature #143
  Implements comment write fe #142

# Conflicts:
#	src/main/java/com/wannaone/woowanote/dto/NoteBookDto.java
#	src/main/java/com/wannaone/woowanote/support/ErrorMessage.java
#	src/main/java/com/wannaone/woowanote/web/ApiNoteBookController.java
#	src/main/resources/messages.properties
#	src/main/resources/static/css/main/notebook-list.css
#	src/test/java/com/wannaone/woowanote/web/ApiNoteBookAcceptanceTest.java

Commit:
Refactor

코드리뷰 반영 기존 비교 메소드 제거 후 id값만 비교하도록 변경

Commit:
Merge branch 'develop' into feature/notebook-delete

* develop:
  Implements note parent notebook update (#149)

# Conflicts:
#	src/main/java/com/wannaone/woowanote/domain/NoteBook.java

Commit:
Implements notebook delete feature #138

노트북 삭제 아이콘, hover시에만 나타나도록 수정

Commit:
Fix bug

노트북 삭제 후 노트북 추가시 2개씩 추가되는 문제 수정(event가 여러번 할당되면서 생긴 문제)
노트북 목록 클릭 시 변경되지 않는 문제 해결, span -> li 선택되도록 수정

Commit:
Fix bug

선택된 노트북 삭제 시 노트 리스트 업데이트 안되던 문제 수정
